packages/data-sdk/src/Device.ts

```diff
@@ -636,19 +634,31 @@ export class Device extends EventEmitter implements IRealtimeDevice {
   }
 
   async stopRealtimeConnection() {
-    if (this.rtcClient && this.remoteDevicePeerId) {
+    let throwNotStartedError = false;
+
+    if (this.rtcClient) {
       this.stopConnectionMonitoring();
-      await this.rtcClient.disconnect(this.remoteDevicePeerId);
-      this.remoteDevicePeerId = null;
-    } else {
+
+      if (this.remoteDevicePeerId) {
+        await this.rtcClient.disconnect(this.remoteDevicePeerId);
+        this.remoteDevicePeerId = null;
+      } else {
+        throwNotStartedError = true;
+      }
+
+      await this.rtcClient.shutdown();
```

Comment:
Wondering if this could cause any issues if the rtcClient _kinda_ shuts down but fails. Then we would not set `this.rtcClient = undefined` and could end up with a broken client?

Reply:
good call/catch, I'll switch it to a try/finally.